src/entities/inputValidator/inputValidator.ts

```diff
@@ -91,4 +95,17 @@ export class InputValidator {
         if (chainId in ChainId) return;
         throw new Error(`Unsupported ChainId: ${chainId}`);
     }
+
+    static validateBuildCallWithPermit2(input: {
```

Comment:
This is not following the pattern, but it simplifies the implementation significantly.
To be honest, I don't like the way InputValidator is currently implemented. Seems to be more complicated than it needs to be.
In any case, I think it's ok as it, but let me know what you think

Reply:
Definitely agree `InputValidator` could be done better and I like the static method pattern you're proposing here. Much easier to read :+1:

Reply:
I don't think I fully appreciate the complexity of following the old pattern here but open to the change if we think it improves things.
If we want to move to a static refactor lets make an issue for it.

Reply:
This buildCallWithPermit input validation is also used by swaps, which don't follow the InputValidator structure. 
The static alternative was an attempt to get around the overhead of that structure and still keep things close together.
There is already an issue about refactoring input validators [here](https://github.com/balancer/b-sdk/issues/280). We could/should expand on that and try to find a new design that is simpler/easier to follow and also a bit more flexible.
It doesn't even need to follow that static pattern if we find another solution that works better considering the whole implementation.
Let me know if you think that's not enough reason to break pattern and I'll adjust over here until we spend proper time on this later on.

Reply:
All good for now I think.